pkg/dbutil/dbutil.go

```diff
@@ -5,13 +5,61 @@ import (
 	"bytes"
 	"database/sql"
 	"errors"
+	"fmt"
 	"io"
 	"net/url"
 	"os/exec"
 	"strings"
 	"unicode"
 )
 
+// DetailedSQLError contains an SQL error and also the line, column and position of the error
+//
+// This was initially created to work around deficiency in the PostgreSQL drivers for Go where
+// this kind of information must be manually computed
+type DetailedSQLError struct {
+	SQLError error
+	Line     int
+	Column   int
+	Position int
+}
+
+var _ error = new(DetailedSQLError)
```

Comment:
what is this line for?

Reply:
explicitly asserts at compile-time that DetailedSQLError satisfies the `error` interface.
https://stackoverflow.com/a/34619122/5013410